tensorrtllm/convert_checkpoint.py

```diff
@@ -0,0 +1,441 @@
+# SPDX-FileCopyrightText: Copyright (c) 2022-2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import argparse
+import json
+import os
+import time
+
+import numpy as np
+import torch
+from safetensors.torch import save_file
+
+import tensorrt_llm
+from tensorrt_llm.functional import LayerNormPositionType, LayerNormType
+from tensorrt_llm.models.convert_utils import weight_only_quantize_dict
+from tensorrt_llm.quantization import QuantAlgo
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--model_dir', type=str, default="assets")
```

Comment:
Can these model checkpoints be on the Hub and directly downloaded from there?

Reply:
The convert_checkpoint.py accepts large-v3.pt and generates trt-llm checkpoints. Which one you would like put on the hub?

Reply:
https://huggingface.co/yuekai/whisper_turbo_and_distill_tllm_checkpoint/tree/main, like this one, if we store the converted checkpoints, we could remove the convert_checkpoint.py file.

Reply:
Yes exactly like the last one is okay and then we can remove the `convert_checkpoint.py` too. I'd recommend uploading each model in a seperate model repo. We can also move them over to `nvidia` org.

Reply:
Done.